Automattic/jetpack
# Override premium block icons

Fixes #16694

#### Changes proposed in this Pull Request:

* Adding an icon indicator in core blocks that are premium eg `core/video` is not possible by altering the source icon code cause we don't have access to it. Therefore, there was a need to override the default icon dynamically. Since we would do this for the premium `core` blocks, I decided to make it more abstract so it includes all premium blocks. Eventually
https://github.com/Automattic/jetpack/blob/da2648cfd0d29e76590d9b976a29c21f78c8fd8c/extensions/shared/premium-blocks/index.js#L22
will return all premium blocks (jetpack, FSE, core), so we are just injecting the icon in the default block icon when that block is premium

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Apply these changes to your Sandbox
* Sandbox a dotcom Simple site
* Go to post edit 
* You should be able to see the icon indicator in all jetpack paid blocks

![image](https://user-images.githubusercontent.com/12430020/89224659-842b2780-d5e1-11ea-83e9-847655d5ce38.png)


#### Proposed changelog entry for your changes:

* Dynamically inject an indicator in premium block icons
